manifests/params.pp

```diff
@@ -14,390 +14,65 @@
-  } else {
-    $safe_manage_client = $manage_client
-  }
-
-  $snmp_service_enable = getvar('::snmp_service_enable')
-  if $snmp_service_enable {
-    $service_enable =  $::snmp_service_enable
-  } else {
-    $service_enable =  true
-  }
-  if is_string($service_enable) {
-    $safe_service_enable = str2bool($service_enable)
-  } else {
-    $safe_service_enable = $service_enable
-  }
-
-  $snmp_service_hasstatus = getvar('::snmp_service_hasstatus')
-  if $snmp_service_hasstatus {
-    $service_hasstatus =  $::snmp_service_hasstatus
-  } else {
-    $service_hasstatus =  true
-  }
-  if is_string($service_hasstatus) {
-    $safe_service_hasstatus = str2bool($service_hasstatus)
-  } else {
-    $safe_service_hasstatus = $service_hasstatus
-  }
-
-  $snmp_service_hasrestart = getvar('::snmp_service_hasrestart')
-  if $snmp_service_hasrestart {
-    $service_hasrestart =  $::snmp_service_hasrestart
-  } else {
-    $service_hasrestart =  true
-  }
-  if is_string($service_hasrestart) {
-    $safe_service_hasrestart = str2bool($service_hasrestart)
-  } else {
-    $safe_service_hasrestart = $service_hasrestart
-  }
-
-  $snmp_trap_service_enable = getvar('::snmp_trap_service_enable')
-  if $snmp_trap_service_enable {
-    $trap_service_enable =  $::snmp_trap_service_enable
-  } else {
-    $trap_service_enable =  false
-  }
-  if is_string($trap_service_enable) {
-    $safe_trap_service_enable = str2bool($trap_service_enable)
-  } else {
-    $safe_trap_service_enable = $trap_service_enable
-  }
-
-  $snmp_trap_service_hasstatus = getvar('::snmp_trap_service_hasstatus')
-  if $snmp_trap_service_hasstatus {
-    $trap_service_hasstatus =  $::snmp_trap_service_hasstatus
-  } else {
-    $trap_service_hasstatus =  true
-  }
-  if is_string($trap_service_hasstatus) {
-    $safe_trap_service_hasstatus = str2bool($trap_service_hasstatus)
-  } else {
-    $safe_trap_service_hasstatus = $trap_service_hasstatus
-  }
-
-  $snmp_trap_service_hasrestart = getvar('::snmp_trap_service_hasrestart')
-  if $snmp_trap_service_hasrestart {
-    $trap_service_hasrestart =  $::snmp_trap_service_hasrestart
-  } else {
-    $trap_service_hasrestart =  true
-  }
-  if is_string($trap_service_hasrestart) {
-    $safe_trap_service_hasrestart = str2bool($trap_service_hasrestart)
-  } else {
-    $safe_trap_service_hasrestart = $trap_service_hasrestart
-  }
-
-  $snmpv2_enable =  true
+  $agentaddress = [ 'udp:127.0.0.1:161', 'udp6:[::1]:161' ]
+  $master = false
+  $agentx_perms = undef
+  $agentx_ping_interval = undef
+  $agentx_socket = undef
+  $agentx_timeout = 1
+  $agentx_retries = 5
+  $snmptrapdaddr = [ 'udp:127.0.0.1:162', 'udp6:[::1]:162' ]
+  $ro_community = 'public'
+  $ro_community6 = 'public'
+  $rw_community = undef
+  $rw_community6 = undef
+  $ro_network = '127.0.0.1'
+  $ro_network6 = '::1'
+  $rw_network = '127.0.0.1'
+  $rw_network6 = '::1'
+  $contact = 'Unknown'
+  $location = 'Unknown'
+  $sysname = $::fqdn
```

Comment:
`fact('networking.fqdn')` or `$fact['networking']['fqdn']` instead of `$::fqdn`?

Reply:
Let's do fact handling in a separate PR